src/modules/logicModule/LM_PC_FundingPot_v1.sol

```diff
@@ -154,40 +196,146 @@ contract LM_PC_FundingPot_v1 is
         )
     {
         Round storage round = rounds[roundId_];
-        AccessCriteria storage accessCriteria = round.accessCriterias[id_];
+        AccessCriteria storage accessCriteria =
+            round.accessCriterias[accessCriteriaId_];
+
+        if (accessCriteria.accessCriteriaType == AccessCriteriaType.OPEN) {
+            return (
+                true,
+                accessCriteria.nftContract,
+                accessCriteria.merkleRoot,
+                true
+            );
+        } else if (accessCriteria.accessCriteriaType == AccessCriteriaType.LIST)
+        {
+            return (
+                false,
+                accessCriteria.nftContract,
+                accessCriteria.merkleRoot,
+                true
+            );
+        } else {
+            return (
+                false,
+                accessCriteria.nftContract,
+                accessCriteria.merkleRoot,
+                false
+            );
+        }
+    }
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    function getRoundAccessCriteriaPrivileges(
+        uint64 roundId_,
+        uint8 accessCriteriaId__
+    )
+        external
+        view
+        returns (
+            uint personalCap_,
+            bool overrideContributionSpan_,
+            uint start_,
+            uint cliff_,
+            uint end_
+        )
+    {
+        Round storage round = rounds[roundId_];
+        AccessCriteria storage accessCriteria =
+            round.accessCriterias[accessCriteriaId__];
+
+        if (accessCriteria.accessCriteriaType == AccessCriteriaType.OPEN) {
+            return (0, false, 0, 0, 0);
+        }
+
+        // Store the privileges in a local variable to reduce stack usage.
+        AccessCriteriaPrivileges storage privileges =
+            roundItToAccessCriteriaIdToPrivileges[roundId_][accessCriteriaId__];
 
-        isOpen = (accessCriteria.accessCriteriaType == AccessCriteriaType.OPEN);
         return (
-            isOpen,
-            accessCriteria.nftContract,
-            accessCriteria.merkleRoot,
-            accessCriteria.allowedAddresses
+            privileges.personalCap,
+            privileges.overrideContributionSpan,
+            privileges.start,
+            privileges.cliff,
+            privileges.end
         );
     }
 
     /// @inheritdoc ILM_PC_FundingPot_v1
-    function getRoundCount() external view returns (uint64 roundCount_) {
-        return nextRoundId;
+    function getRoundCount() external view returns (uint64) {
+        return roundCount;
     }
 
     /// @inheritdoc ILM_PC_FundingPot_v1
-    function getRoundAccessCriteriaCount(uint64 roundId_)
-        public
-        view
-        returns (uint8 accessCriteriaCount_)
-    {
-        return roundIdtoAccessId[roundId_];
+    function isRoundClosed(uint64 roundId_) external view returns (bool) {
+        return roundIdToClosedStatus[roundId_];
+    }
+
+    /// @inheritdoc ILM_PC_FundingPot_v1
+    function getUserEligibility(
```

Comment:
Right now this function is a bit problematic. It returns this RoundUserEligibility struct which doesn't really help the UI to understand if a user can contribute to a given accessCriteriaId.

I think this function should let the user specify
- roundId
- accessCriteriaId
- user
- merkleProof

and return _how much the user can still contribute_ to the accessCriteriaId..